bazelisk build //kythe/release broken on macos

```
$ bazel build //kythe/release
...
ERROR: .../kythe/kythe/cxx/verifier/BUILD:25:1: C++ compilation of rule '//kythe/cxx/verifier:lexparse' failed (Exit 1) wrapped_clang failed: error executing command external/local_config_cc/wrapped_clang '-D_FORTIFY_SOURCE=1' -fstack-protector -fcolor-diagnostics -Wall -Wthread-safety -Wself-assign -fno-omit-frame-pointer -O0 -DDEBUG '-std=c++11' -iquote . -iquote ... (remaining 37 argument(s) skipped)

Use --sandbox_debug to see verbose messages from the sandbox
bazel-out/darwin-fastbuild/bin/kythe/cxx/verifier/lexer.yy.cc:867:2: error: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
        register yy_state_type yy_current_state;
        ^~~~~~~~~
bazel-out/darwin-fastbuild/bin/kythe/cxx/verifier/lexer.yy.cc:868:2: error: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
        register char *yy_cp, *yy_bp;
        ^~~~~~~~~
bazel-out/darwin-fastbuild/bin/kythe/cxx/verifier/lexer.yy.cc:868:2: error: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
        register char *yy_cp, *yy_bp;
        ^~~~~~~~~
bazel-out/darwin-fastbuild/bin/kythe/cxx/verifier/lexer.yy.cc:869:2: error: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
        register int yy_act;
        ^~~~~~~~~
bazel-out/darwin-fastbuild/bin/kythe/cxx/verifier/lexer.yy.cc:938:4: error: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
                        register YY_CHAR yy_c = yy_ec[YY_SC_TO_UI(*yy_cp)];
                        ^~~~~~~~~
bazel-out/darwin-fastbuild/bin/kythe/cxx/verifier/lexer.yy.cc:1345:6: error: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
        register char *dest = YY_CURRENT_BUFFER_LVALUE->yy_ch_buf;
        ^~~~~~~~~
bazel-out/darwin-fastbuild/bin/kythe/cxx/verifier/lexer.yy.cc:1346:2: error: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
        register char *source = (yytext_ptr);
        ^~~~~~~~~
bazel-out/darwin-fastbuild/bin/kythe/cxx/verifier/lexer.yy.cc:1347:2: error: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
        register int number_to_move, i;
        ^~~~~~~~~
bazel-out/darwin-fastbuild/bin/kythe/cxx/verifier/lexer.yy.cc:1347:2: error: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
        register int number_to_move, i;
        ^~~~~~~~~
bazel-out/darwin-fastbuild/bin/kythe/cxx/verifier/lexer.yy.cc:1485:2: error: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
        register yy_state_type yy_current_state;
        ^~~~~~~~~
bazel-out/darwin-fastbuild/bin/kythe/cxx/verifier/lexer.yy.cc:1486:2: error: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
        register char *yy_cp;
        ^~~~~~~~~
bazel-out/darwin-fastbuild/bin/kythe/cxx/verifier/lexer.yy.cc:1494:3: error: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
                register YY_CHAR yy_c = (*yy_cp ? yy_ec[YY_SC_TO_UI(*yy_cp)] : 1);
                ^~~~~~~~~
bazel-out/darwin-fastbuild/bin/kythe/cxx/verifier/lexer.yy.cc:1523:2: error: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
        register int yy_is_jam;
        ^~~~~~~~~
bazel-out/darwin-fastbuild/bin/kythe/cxx/verifier/lexer.yy.cc:1525:2: error: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
        register char *yy_cp = (yy_c_buf_p);
        ^~~~~~~~~
bazel-out/darwin-fastbuild/bin/kythe/cxx/verifier/lexer.yy.cc:1527:2: error: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
        register YY_CHAR yy_c = 1;
        ^~~~~~~~~
15 errors generated.
Target //kythe/release:release failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 4.665s, Critical Path: 3.51s
INFO: 13 processes: 11 darwin-sandbox, 2 worker.
FAILED: Build did NOT complete successfully
```

I aliased bazelisk to bazel.

--- a/kythe/web/site/getting-started.md
+++ b/kythe/web/site/getting-started.md
@@ -33,7 +33,7 @@ Kythe relies on the following external dependencies:
 * bison-3.0.4
 * clang >= 8
 * [docker](https://www.docker.com/) (for release images `//kythe/release/...` and `//buildtools/docker`)
-* flex-2.5
+* flex-2.6
 * go >= 1.7
 * graphviz
 * jdk >= 8

--- a/tools/build_rules/lexyacc/lexyacc.bzl
+++ b/tools/build_rules/lexyacc/lexyacc.bzl
@@ -1,3 +1,5 @@
+load("@bazel_skylib//lib:versions.bzl", "versions")
+
 def genlex(name, src, out, includes = []):
     """Generate a C++ lexer from a lex file using Flex.
 
@@ -83,10 +85,22 @@ def lexyacc_toolchain(name, lex, yacc):
         toolchain_type = "@io_kythe//tools/build_rules/lexyacc:toolchain_type",
     )
 
-def _local_lexyacc(repository_ctx):
+def _check_flex_version(repository_ctx, min_version):
     flex = repository_ctx.which("flex")
     if flex == None:
         fail("Unable to find flex binary")
+    flex_result = repository_ctx.execute([flex, "--version"])
+    if flex_result.return_code:
+        fail("Unable to determine flex version: " + flex_result.stderr)
+    flex_version = flex_result.stdout.split(" ")
+    if len(flex_version) < 2:
+        fail("Too few components in flex version: " + flex_result.stdout)
+    if not versions.is_at_least(min_version, flex_version[1]):
+        fail("Flex too old (%s < %s)" % (flex_version[1], min_version))
+    return flex
+
+def _local_lexyacc(repository_ctx):
+    flex = _check_flex_version(repository_ctx, "2.6")
     bison = repository_ctx.os.environ.get("BISON", repository_ctx.which("bison"))
     if not bison:
         fail("Unable to find bison binary")

